libs/datagrid/src/lib/cells/market-name-cell.tsx

```diff
@@ -0,0 +1,40 @@
+import type { MouseEvent } from 'react';
+import { useCallback } from 'react';
+import get from 'lodash/get';
+import { Link } from 'react-router-dom';
+
+const MARKET_PATH = 'markets';
+
+interface MarketNameCellProps {
+  value?: string;
+  data?: { id?: string; marketId?: string; market?: { id: string } };
+  idPath?: string;
+  onMarketClick?: (marketId: string, metaKey?: boolean) => void;
+}
+
+export const MarketNameCell = ({
+  value,
+  data,
+  idPath,
+  onMarketClick,
+}: MarketNameCellProps) => {
+  const id = data ? get(data, idPath ?? 'id', 'all') : '';
+  const marketLink = `/${MARKET_PATH}/${id}`;
+  const handleOnClick = useCallback(
+    (ev: MouseEvent<HTMLAnchorElement>) => {
+      ev.preventDefault();
+      ev.stopPropagation();
+      if (onMarketClick) {
+        onMarketClick(id, ev.metaKey);
+        return;
+      }
+    },
+    [id, onMarketClick]
+  );
+  if (!data) return null;
+  return (
+    <Link to={marketLink} onClick={handleOnClick}>
```

Comment:
chore (blocking): We should remove the dependency on react-router-dom here. I'd suggest just using a span but with appropriate attributes
```suggestion
    <span onClick={handleOnClick} role="link" tabIndex="0">
```